components/CollectivePicker.js

```diff
@@ -151,40 +159,62 @@ class CollectivePicker extends React.PureComponent {
     });
   });
 
-  getAllOptions = memoizeOne((collectivesOptions, customOptions, createdCollectives, creatable, intl) => {
-    let options = collectivesOptions;
-
-    if (createdCollectives.length > 0) {
-      options = [...createdCollectives.map(this.buildCollectiveOption), ...options];
-    }
-
-    if (customOptions && customOptions.length > 0) {
-      options = [...customOptions, ...options];
-    }
-
-    if (creatable) {
-      const isOnlyForUser = isEqual(this.props.types, [CollectiveType.USER]);
-      options = [
-        ...options,
-        {
-          label: isOnlyForUser
-            ? intl.formatMessage(Messages.inviteNew).toUpperCase()
-            : intl.formatMessage(Messages.createNew).toUpperCase(),
-          options: [
-            {
-              label: null,
-              value: null,
-              isDisabled: true,
-              [FLAG_NEW_COLLECTIVE]: true,
-              __background__: 'white',
-            },
-          ],
-        },
-      ];
-    }
-
-    return options;
-  });
+  getAllOptions = memoizeOne(
+    (collectivesOptions, customOptions, createdCollectives, { creatable, invitable, intl, customOptionsPosition }) => {
```

Comment:
We're loosing the advantage of memoization here, the 4th argument being an object built dynamically its reference will be different each time (a call like `fn({})` cannot be memoized).

Two possible solutions:
1. Add a function to generate this object (ie. `generateOptionsParams`) that is itself memoized (so that it returns the same object each time it's called)
2. Pass each argument individually (which creates some complexity because of the numbers of arguments.

My personal fav is 1. The simplest way to test that memoization works is to add a `console.log` at the beginning of `getAllOptions` to make sure it isn't called at each render.

Reply:
Funny fact, all these variables are coming from props, which is directly accessible through `this.props`.

Reply:
@kewitz When memoized, the functions parameters are used as a "cache key". That's why they were passed as parameters, and not accessed directly from `this.props`. With your latest change, the options list will not be refreshed if you change one of the variables taken from the props (`creatable`, `invitable`, `customOptionsPosition`).

I don't think it's a problem at the moment (I don't see a use case for changing these props after the component has been mounted) but it will result in an inconsistent behavior if someones tries to do so in the future.

Reply:
To be honest, I was already taking the memoization into consideration, but you're right, it is wrong to take into consideration that after mounted the component will stay `creatable` or `invitable`.